File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/cmd/IOTCommand.java
package com.shmedo.configlibrary.iot.cmd;

import com.shmedo.configlibrary.ble.interfaces.Validater;
import com.shmedo.configlibrary.iot.enums.IOTCommandType;

/**
 * 命令类，把命令头和命令码和参数组装成相应的命令
 */
public class IOTCommand<T extends Validater> {
    public static final String COMMAND_HEADER = "$cmd=";

    private IOTCommandType commandType;

    private T parameters;

    private String apiKey;

    public IOTCommand(IOTCommandType commandType) {
        this.commandType = commandType;
    }

    public IOTCommand(IOTCommandType commandType, T parameters) {
        this.commandType = commandType;
        this.parameters = parameters;
    }

    @Override
    public String toString() {
//        if (parameters != null)
//            parameters.validate();

        String paraString = parameters == null ? "" : parameters.toString();

        return COMMAND_HEADER + commandType.toString() + "&"
                + paraString;
//                + "&apikey=" + apiKey
//                + "&msgid=" + UUID.randomUUID().toString();
//                + "&&";
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/iot_cmd/IOTCommand.kt
package com.shmedo.lib.device.base.iot_cmd

import com.shmedo.lib.device.base.iot_cmd.enums.IOTCommandType

/**
 * 命令类，把命令头和命令码和参数组装成相应的命令
 */
class IOTCommand<T>(val commandType: IOTCommandType, private val parameters: T? = null) {

    companion object {
        const val COMMAND_HEADER = "\$cmd="
    }

    override fun toString(): String {
        return if (parameters == null) {
            COMMAND_HEADER + commandType
        } else {
            "$COMMAND_HEADER$commandType&$parameters"
        }
    }
}
